kvmagent/ansible/kvm.py

```
@@ -113,6 +113,111 @@ def update_libvirtd_config(host_post_info):
+            verify_cmd = (
+                "openssl verify -CAfile /etc/pki/CA/cacert.pem /etc/pki/libvirt/servercert.pem 2>&1 | grep -q ': OK' && "
+                "openssl verify -CAfile /etc/pki/CA/cacert.pem /etc/pki/libvirt/clientcert.pem 2>&1 | grep -q ': OK'"
+            )
+            (verify_status, _) = run_remote_command(verify_cmd, host_post_info, return_status=True, return_output=True)
+            if verify_status == 0:
+                handle_ansible_info("Libvirt TLS certs already valid on host, skipping", host_post_info, "INFO")
+                return
+        handle_ansible_info("Remote certs incomplete or CA mismatch, re-deploying", host_post_info, "INFO")
+
+    host_ip = host_post_info.host
+    cert_tmp_dir = "/tmp/zstack-libvirt-tls-%s" % host_ip.replace('.', '_')
```

Ruff (0.15.5)

[error] Probable insecure usage of temporary file or directory: "/tmp/zstack-libvirt-tls-%s"

(S108)